test/should_pass/record_refinement.erl

```diff
@@ -78,3 +78,9 @@ two_level2(#two_level1{two_level2 = #two_level2{value = undefined}}) ->
     0;
 two_level2(R1) ->
     R1#two_level1.two_level2#two_level2.value.
+
+-record(not_all_fields_refined, {a, b}).
+-type not_all_fields_refined() :: #not_all_fields_refined{b :: b | c}.
+
+-spec not_all_fields_refined() -> not_all_fields_refined().
```

Comment:
I think it's a tiny bit confusing to use the same name for a function and a type.